components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -312,59 +312,76 @@ private Response processRequest(final Map<String, String> requestAttributes) {
                 String sortOrder = requestAttributes.get(SCIMProviderConstants.SORT_ORDER);
                 String domainName = requestAttributes.get(SCIMProviderConstants.DOMAIN);
 
-                scimResponse = groupResourceManager.listWithGET(userManager, filter, startIndex,
-                        count, sortBy, sortOrder, domainName, attributes, excludedAttributes);
-
+                // Processing count and startIndex in the request.
+                Integer startIndex = processPaginationParamsInRequest(
+                        requestAttributes.get(SCIMProviderConstants.START_INDEX), SCIMProviderConstants.START_INDEX);
+                Integer count = processPaginationParamsInRequest(requestAttributes.get(SCIMProviderConstants.COUNT),
+                        SCIMProviderConstants.COUNT);
+                scimResponse = groupResourceManager
+                        .listWithGET(userManager, filter, startIndex, count, sortBy, sortOrder, domainName, attributes,
+                                excludedAttributes);
             } else if (GET.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.get(id, userManager, attributes, excludedAttributes);
-            } else if (POST.class.getSimpleName().equals(httpVerb) && search.equals("1")){
-                scimResponse = groupResourceManager.listWithPOST(resourceString,userManager);
+            } else if (POST.class.getSimpleName().equals(httpVerb) && search.equals("1")) {
+                scimResponse = groupResourceManager.listWithPOST(resourceString, userManager);
             } else if (POST.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.create(resourceString, userManager, attributes, excludedAttributes);
             } else if (PUT.class.getSimpleName().equals(httpVerb)) {
-                scimResponse =
-                        groupResourceManager.updateWithPUT(id, resourceString, userManager, attributes, excludedAttributes);
+                scimResponse = groupResourceManager
+                        .updateWithPUT(id, resourceString, userManager, attributes, excludedAttributes);
             } else if (PATCH.class.getSimpleName().equals(httpVerb)) {
-                scimResponse =
-                        groupResourceManager.updateWithPATCH(id, resourceString, userManager, attributes, excludedAttributes);
+                scimResponse = groupResourceManager
+                        .updateWithPATCH(id, resourceString, userManager, attributes, excludedAttributes);
             } else if (DELETE.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.delete(id, userManager);
             }
-
             return SupportUtils.buildResponse(scimResponse);
-
         } catch (CharonException e) {
             return handleCharonException(e, encoder);
         }
     }
 
+    /**
+     * Method to process the count and startIndex in the request.
```

Comment:
Shall we change the doc comment, since the method itself is not specific for `count and startIndex`.

Reply:
fixed: df77256698c74c829147f73d00e2f63979199c16